### PR TITLE
Remove unused code

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Ideally we could completely remove the CLRs private native RegEx implementation. It is being used by some eeconfig settings in debug mode. It might be useful to degrade those settings and instead rely on a "contains" semantic instead of the far more complicated regex support.

### CODE REVIEW

**Comment by jkotas** [comment]
> is it being used by some eeconfig settings in debug mode.

What are those settings? The only one that I can see is `pPerfTypesToLog`. I do not think that `pPerfTypesToLog` is actively used for anything. It should be fine to degrade it to contains semantics.

**Comment by AaronRobinsonMSFT** [comment]
> > is it being used by some eeconfig settings in debug mode.
> 
> What are those settings? The only one that I can see is `pPerfTypesToLog`. I do not think that `pPerfTypesToLog` is actively used for anything. It should be fine to degrade it to contains semantics.

@jkotas  It is used in a bunch of settings, all in Debug. Look for `RegexOrExactMatch`. In the most recent commit I proposed using the very simple Rob Pike version from TPOP. It provides some of the functionality so people can still do a little regex. I can also just replace with with the contains. I've no strong preference other than we should get rid of our native Regex version.

File: src/coreclr/vm/eeconfig.cpp [hunk 1]
```diff
@@ -943,6 +942,52 @@ TypeNamesList::TypeNamesList()
     LIMITED_METHOD_CONTRACT;
 }
 
+namespace
+{
+    //
+    // Slight modification of Rob Pike's minimal regex from The Practice of Programming.
+    // https://en.wikipedia.org/wiki/The_Practice_of_Programming
```
Comment (jkotas) [inline src/coreclr/vm/eeconfig.cpp:949]: Does the book mention a license for code fragments?

**Comment by jkotas** [comment]
> In the most recent commit I proposed using the very simple Rob Pike version from TPOP.

Looks good to me.

File: src/coreclr/vm/eeconfig.cpp (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/eeconfig.cpp:949]: I haven't found anything about license in the book itself - written in 1999. The official site does have the following page which I can reference.

https://www.cs.princeton.edu/~bwk/tpop.webpage/code.html

File: src/coreclr/vm/eeconfig.cpp [hunk 2]
```diff
@@ -943,6 +942,52 @@ TypeNamesList::TypeNamesList()
     LIMITED_METHOD_CONTRACT;
 }
 
+namespace
+{
+    //
+    // Slight modification of Rob Pike's minimal regex from The Practice of Programming.
+    // See https://www.cs.princeton.edu/~bwk/tpop.webpage/code.html - 'Grep program from Chapter 9'
```
Comment (jkotas) [inline src/coreclr/vm/eeconfig.cpp:949]: This should get mention in https://github.com/dotnet/runtime/blob/main/THIRD-PARTY-NOTICES.TXT. Something like:
```
License notice for code from The Practice of Programming
-------------------------------

Copyright (C) 1999 Lucent Technologies

Excerpted from 'The Practice of Programming
by Brian W. Kernighan and Rob Pike

You may use this code for any purpose, as long as you leave the copyright notice and book citation attached.
```

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**